update device api default values for `execution_config` (#7697)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Breaking changes 💔</h3>

> * `ExecutionConfig` and `MCMConfig` from `pennylane.devices` are now frozen dataclasses and cannot be mutated without the use of the `dataclass.replace` function. 

Author: ```suggestion
* `ExecutionConfig` and `MCMConfig` from `pennylane.devices` are now frozen dataclasses whose fields should be updated with `dataclass.replace`. 
```

<h3>Breaking changes 💔</h3>

> * The `Device` API now defaults `execution_config : Optional[ExecutionConfig] = None` instead of defaulting

Author: I personally would leave the type hint to highlight that it is indeed an optional kwarg.

> * The `Device` API now defaults `execution_config : Optional[ExecutionConfig] = None` instead of defaulting

Author: Oh I forgot that was added lol. Sorry I'll take your suggestion!

<h3>Breaking changes 💔</h3>

> * `ExecutionConfig` and `MCMConfig` are now frozen dataclasses. This means that their properties cannot be modified in place and instead need to be updated with `dataclasses.replace`.
>   [(#7697)](https://github.com/PennyLaneAI/pennylane/pull/7697)
> 

Reviewer: ```suggestion
* The `Device` API now defaults `execution_config : Optional[ExecutionConfig] = None` instead of defaulting
  to `qml.devices.DefaultExecutionConfig`.  This prevents accidental mutation of a global variable that
  would alter the default behaviour on later executions.
  
  This change applies to all methods that accept an `execution_config`. The new call signature for `execute`
  is:
  ```
  def execute(self, circuits: QuantumScriptOrBatch, execution_config: Optional[ExecutionConfig] =None):
  ```

  [(#7697)](https://github.com/PennyLaneAI/pennylane/pull/7697/)

* `ExecutionConfig` and `MCMConfig` are now frozen dataclasses. This means that their properties cannot be modified in place and instead need to be updated with `dataclasses.replace`.
  [(#7697)](https://github.com/PennyLaneAI/pennylane/pull/7697)

```

Or something like that.
